src/ansys/mapdl/core/mapdl.py

```diff
@@ -3204,9 +3204,29 @@ def run(
         if "\n" in command or "\r" in command:
             raise ValueError("Use ``input_strings`` for multi-line commands")
 
+        # Checking kwargs
+        verbose = kwargs.pop("verbose", False)
+        save_fig = kwargs.pop("savefig", False)
+
         # check if we want to avoid the current non-interactive context.
         avoid_non_interactive = kwargs.pop("avoid_non_interactive", False)
 
+        # Checking if there is any non-used keyword argument. If there is, it
+        # might be because we wrote a wrong argument name.
+        #
+        # Removing empty string kwargs
+        for key, value in list(kwargs.items()):
+            if value == "":
+                kwargs.pop(key)
+
+        if kwargs:
+            raise ValueError(
+                "The following keyword arguments are not used by 'mapdl.run':\n"
+                f"{', '.join(kwargs.keys())}\n"
+                "Make sure you are using the intended keyword arguments in all\n"
+                "your mapdl method calls (for instance 'mapdl.something')."
+            )
```

Comment:
It seems OK--but I'm not that well versed in the tagging used in error messages. In regular text, a single back tack ends up rending the text in italics, rather than monospace. You need double backticks to render in monospace--which should be used for code entities, such as methods, file paths, and the names of folders and files.

Reply:
@Gryfenfer97 @clatapie any feedback?

Reply:
I don't really know if we should specify in this error to check in all mapdl method calls. It could be confusing for the developer reading the error.

Also, isn't it too extreme to throw an error for that ? Maybe a warning would work too

Reply:
> I don't really know if we should specify in this error to check in all mapdl method calls. It could be confusing for the developer reading the error.

Mmhh.... the thing is we can only check if we are not using a kwarg at `mapdl.run`, the other functions, the arguments name can be anything. But getting an error that say `"mapdl.run" is not using XXX kwarg` might be misleading. I guess that the user should be able to use the traceback to see in which line of HIS/HER code is the error... I guess we could just say that `XXX kwarg is not used` and not mentioning `mapdl.run` at all, so the user will check first its code.


> Also, isn't it too extreme to throw an error for that ? Maybe a warning would work too

mmhhh.... If you pass a kwarg which is not used.... what is the best behaviour? raise an error or a warning? 
I think the user should be aware that it is not used... warning might be enough... I dont know.... Pinging @koubaa  @RobPasMue and @jorgepiloto for feedback.

Reply:
In this case, raising an error has the advantage of giving the user the traceback so that he can find out which function caused the error.

Reply:
Ok... I'm going to change the message and use warnings. Thank you all!

Reply:
I agree with using warnings. Also, I think that **removing the `mapdl.run`** mention in the text would help the user to suspect an error in his/her own code.
If a user is not familiar with pymapdl, he/she will not know that a call to `mapdl.dl` (for instance) involves the `mapdl.run` call in the background. He/She might be confused that a warning related to a function that might not be present in the initial code appears.

Reply:
Thank you @clatapie I will do that :)